frame/support/src/lib.rs

```diff
@@ -1550,17 +1554,17 @@ pub mod pallet_prelude {
 ///
 /// ```
 /// #[frame_support::pallet]
-/// // NOTE: Example is name of the pallet, it will be used as unique identifier for storage
+/// // NOTE: The name of the pallet will be provided by construct_runtime, and will be used as
+/// // unique identifier for storage. It is not defined in the pallet itself.
 /// pub mod pallet {
 /// 	use frame_support::pallet_prelude::*; // Import various types used in pallet definition
-/// 	use frame_system::pallet_prelude::*; // OriginFor helper type for implementing dispatchables.
+/// 	use frame_system::pallet_prelude::*; // Import some system helper types.
 ///
 /// 	type BalanceOf<T> = <T as Config>::Balance;
 ///
 /// 	// Define the generic parameter of the pallet
-/// 	// The macro checks trait generics: is expected none or `I = ()`.
-/// 	// The macro parses `#[pallet::constant]` attributes: used to generate constant metadata,
-/// 	// expected syntax is `type $IDENT: Get<$TYPE>;`.
+/// 	// The macro parses `#[pallet::constant]` attributes, and use them to generate constant
+/// 	// metadata,
```

Comment:
```suggestion
/// 	// The macro parses `#[pallet::constant]` attributes, and uses them to generate constant
/// 	// metadata,
```

I wonder if rather than "constant" it'd be clearer to use "static" or "immutable" here. To me this could be misunderstood as if the macro generates metadata **continuously**.

Reply:
at some point metadata is generated everytime `sp_api::Metadata::metadata` (i.e. metadata runtime api) gets called by the client.
I think in the context of metadata, "constant" is not less or more meaningful.
Maybe better "constant_metadata" 🤔